### PR TITLE
Refactor dependency overhall

### PR DESCRIPTION
**Pull request by msecret**

### CODE REVIEW

File: build/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+gulp.task('svg-sprite', function() {
+  return gulp.src(base + './src/img/**/*.svg')
+    .pipe(svgSprite(config))
+    .pipe(gulp.dest(base + './img'));
+});
+
+gulp.task('default', function() {
+  // place code for your default task here
```
Comment (jcscottiii) [inline build/gulpfile.js]: replace this comment with what the default task should do at a high level

File: build/gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+var config = {
+  mode: {
+    symbol: {
+      dest: '', 
+      sprite: 'icons.svg'
+    }
+  }
+};
+
+gulp.task('svg-sprite', function() {
```
Comment (jcscottiii) [inline build/gulpfile.js:17]: comment for this task

File: build/img_spriting.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (jcscottiii) [inline build/img_spriting.js:5]: this is for my own ignorance lol. a comment here for what the file does.

File: build/img_spriting.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      return Promise.reject();
+    }
+
+    return Promise.resolve();
+  },
+  stylesheetPath: './css',
+  spritePath: './img'
+};
+
+postcss([ require('postcss-sprites').default(opts) ])
```
Comment (jcscottiii) [inline build/img_spriting.js:23]: more for my ignorance, how are you using these `opts`.

File: build/img_spriting.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    .process(css, { 
+      from: './css/cg_style.css',
```
Comment (jcscottiii) [inline build/img_spriting.js:25]: same from and to?

File: build/img_spriting.js (hunk 4 above)
Comment (msecret) [inline build/img_spriting.js:23]: It's being passed to `require('postcss-sprites').default(opts)`.

File: build/img_spriting.js (hunk 5 above)
Comment (msecret) [inline build/img_spriting.js:25]: Yea it's overwriting the file.

File: src/css/components/icon.scss [hunk 6]
```diff
@@ -2,14 +2,26 @@
...
 .icon {
-  max-height: 3.1rem;
+  display: inline-block;
+  height: 100%;
+  fill: transparent;
+  max-height: 2rem;
+  max-width: 2rem;
+  stroke: currentColor;
+  position: relative;
+  top: -0.0625em;
```
Comment (jcscottiii) [inline src/css/components/icon.scss]: em -> rem

**Pull request merged**